index.js

```diff
@@ -4,6 +4,13 @@ let chokidar = require("chokidar");
 let EventEmitter = require("events");
 let path = require("path");
 
+class TooManyFilesError extends Error {
```

Comment:
Somewhat belatedly: I'd prefer for this utility class to reside at the bottom of the module.

Reply:
Feel free to move it 😉